### DISCUSSION TITLE
导航菜单问题：非主菜单的其他菜单获取方式

### DISCUSSION THREAD

**Discussion by holwell** [halo-dev/halo#7145] [Q&A] [answered]
### 前置条件

- [x] 已经在 [Issues](https://github.com/halo-dev/halo/issues) 列表中搜索了相关问题。
- [x] 这是和 Halo 相关的特性，如果是非项目本身的问题，建议在 [Discussions](https://github.com/halo-dev/halo/discussions) 提交。
- [x] 如果是插件和主题特性建议，请在对应的插件和主题仓库提交。

### 你当前使用的版本

2.20.0

### 描述一下此特性

* 文档地址：
  https://docs.halo.run/developer-guide/theme/finder-apis/menu
* 问题：
  只能通过 menu 的 metadata.name 来获取非主菜单的其他菜单吗？如果有其他方式获取菜单组，希望能在文档中补充，非常感谢！
* 建议:
  毕竟 metadata.name 是一个内部数据，无法直接暴露给主题开发者，希望在新增菜单的位置给一个自定义 metadata.name 的输入框，或者像 [链接](https://github.com/halo-sigs/plugin-links?tab=readme-ov-file#linkgroupvo) 插件一样，给一个包含所有菜单信息组的 `group` 变量。

**Comment by ruibaby** [comment]
你应该让用户在主题设置中选择，https://docs.halo.run/developer-guide/form-schema#menuselect

**Comment by ruibaby** [comment]
相关问题：https://github.com/halo-dev/halo/issues/7090#issuecomment-2505628852

**Comment by ruibaby** [comment, marked as answer]
> 如果有其他方式获取菜单组，希望能在文档中补充，非常感谢！

菜单的文档应该是忽略掉了，其他类似的应该有这样写：

<img width="714" alt="image" src="https://github.com/user-attachments/assets/d3a653b5-8adb-48d7-97ec-c9c64ff547cb" />

**Reply by holwell** [reply]
> > 如果有其他方式获取菜单组，希望能在文档中补充，非常感谢！
> 
> 菜单的文档应该是忽略掉了，其他类似的应该有这样写：
> 
> <img alt="image" width="714" src="https://private-user-images.githubusercontent.com/21301288/396009285-d3a653b5-8adb-48d7-97ec-c9c64ff547cb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzMzYzODUsIm5iZiI6MTczNDMzNjA4NSwicGF0aCI6Ii8yMTMwMTI4OC8zOTYwMDkyODUtZDNhNjUzYjUtOGFkYi00OGQ3LTk3ZWMtYzljNjRmZjU0N2NiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE2VDA4MDEyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZlNzQ0NWI1NDk3NGQ1N2I2ZWQyYjExMDZjOTFmNGUwNzc1OWNlMTg0ZmNiZTE2YTZmOTJmMWIwY2VkMWE3YzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.fWZgzMsmvZO7SjWX8oY7ZbQ9wZA0h0qVDw9J7pYamng">

非常感谢🙏
